### PR TITLE
Reproduce revision inconsistency issue

### PR DESCRIPTION
**Pull request by ahrtr**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/linearizability/traffic.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type Traffic interface {
 type putGetTraffic struct{}
 
 func (t putGetTraffic) Run(ctx context.Context, c *recordingClient, limiter *rate.Limiter) {
-	maxOperationsPerClient := 1000000
+	maxOperationsPerClient := 10
```
Comment (serathius) [inline tests/linearizability/traffic.go:36]: Any reason to set this? This number is used to ensure each client write has unique value. This is done by shifting starting id value for each client and limiting number of writes. For example first client writes values `0...1000000`, second client writes `1000000...2000000` and so forth. 

Lowering this number just results in clients making 10 requests and finishing.

File: tests/linearizability/linearizability_test.go [hunk 2]
```diff
@@ -94,30 +78,31 @@ func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProce
...
 	go func() {
-		defer cancel()
 		err := triggerFailpoints(ctx, t, clus, failpoint)
 		if err != nil {
 			t.Error(err)
 		}
 	}()
 	operations := simulateTraffic(ctx, t, clus, traffic)
-	clus.Close()
+	time.Sleep(10 * time.Second)
```
Comment (serathius) [inline tests/linearizability/linearizability_test.go:92]: I would not recommend removing cluster.Stop here as this is needed to saftly copy out the data dir.

File: tests/linearizability/linearizability_test.go [hunk 3]
```diff
@@ -94,30 +78,31 @@ func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProce
...
 
 	linearizable, info := porcupine.CheckOperationsVerbose(etcdModel, operations, 0)
 	if linearizable != porcupine.Ok {
-		t.Error("Model is not linearizable")
+		t.Error("###### Model is not linearizable")
 	}
 
-	path, err := filepath.Abs(filepath.Join(resultsDirectory, strings.Replace(t.Name(), "/", "_", -1)+".html"))
+	path, err := filepath.Abs(filepath.Join("/Users/wachao/tmp/etcd/test/", strings.Replace(t.Name(), "/", "_", -1)+".html"))
```
Comment (serathius) [inline tests/linearizability/linearizability_test.go:99]: You can just use `RESULTS_DIR` environment variable

**Comment by serathius** [comment]
Strange that you are able to reproduce issue locally as I don't see any changes that influences the test itself.

**Comment by ahrtr** [comment]
This PR isn't targeting for merge. It's just show you how I reproduced the issue with simplified test case.

**Pull request closed without merging**